Mathlib/RingTheory/Ideal/Over.lean

```diff
@@ -430,3 +430,98 @@ lemma map_eq_top_iff {R S} [CommRing R] [CommRing S]
 end IsDomain
 
 end Ideal
+
+section ideal_lies_over
+
+variable (A : Type*) [CommSemiring A] {B : Type*} [Semiring B] [Algebra A B]
+  (P : Ideal B) (p : Ideal A)
+
+/-- The ideal obtained by pulling back the ideal `P` from `B` to `A`. -/
+abbrev Ideal.under : Ideal A := Ideal.comap (algebraMap A B) P
+
+theorem Ideal.under_def : Ideal.under A P = Ideal.comap (algebraMap A B) P := rfl
+
+instance Ideal.IsPrime.under [hP : P.IsPrime] : (Ideal.under A P).IsPrime :=
+  hP.comap (algebraMap A B)
+
+/-- If `P` is a maximal ideal of `B`, then the intersection of `P` and `A` is also
+a maximal ideal. -/
+instance Ideal.IsMaximal.under (A : Type*) [CommRing A] {B : Type*} [CommRing B] (P : Ideal B)
+    [P.IsMaximal] [Algebra A B] [Algebra.IsIntegral A B] : (Ideal.under A P).IsMaximal :=
+  isMaximal_comap_of_isIntegral_of_isMaximal P
+
+variable {A}
+
+/-- `P` lies over `p` if `p` is the preimage of `P` of the `algebraMap`. -/
+class ideal_lies_over : Prop where
+  over : p = Ideal.comap (algebraMap A B) P
+
+@[inherit_doc] infix : 50 "lies_over" => ideal_lies_over
```

Comment:
The naming convention requires classes to be in `PascalCase`, i.e. `IdealLiesOver` in this case. I would suggest `Ideal.LiesOver` instead so that you could write `P.LiesOver p` so that you wouldn't need the infix.

Reply:
Also please read the rest of [this guide](https://leanprover-community.github.io/contribute/naming.html) and change the lemma names below accordingly.